src/extensions/default/InlineColorEditor/main.js

```diff
@@ -39,41 +41,43 @@ define(function (require, exports, module) {
      * @return {?{color:String, marker:TextMarker}}
      */
     function prepareEditorForProvider(hostEditor, pos) {
-        var colorRegEx, cursorLine, match, sel, start, end, endPos, marker;
-
-        sel = hostEditor.getSelection();
-        if (sel.start.line !== sel.end.line) {
-            return null;
-        }
-
-        colorRegEx = new RegExp(ColorUtils.COLOR_REGEX);
+        var cursorLine, cssPropertyName, marker, endPos, end, semiColon, colorValue, hash, colon;
         cursorLine = hostEditor.document.getLine(pos.line);
 
-        // Loop through each match of colorRegEx and stop when the one that contains pos is found.
-        do {
-            match = colorRegEx.exec(cursorLine);
-            if (match) {
-                start = match.index;
-                end = start + match[0].length;
+        // Make a copy of cursorLine after removing spaces and ":" so that we can check for it in properties
+        cssPropertyName = cursorLine.split(':')[0].trim();
+
+        if (properties[cssPropertyName]) {
```

Comment:
Flip your logic here and return early, so you don't have to indent a huge block of code:

```js
if (!properties[cssPropertyName]) {
    return null;
}

// Now do the rest of your code here, without indenting
```